apps/dashboard/src/components/editor/toolbar.js

```diff
@@ -24,22 +24,30 @@ import { ToolbarButton } from './styles/button';
 const Toolbar = ( { project } ) => {
 	const { saveEditorContent } = useDispatch( STORE_NAME );
 
-	const [ canRestoreDraft, editorContent, isSaving, isSaved ] = useSelect(
-		( select ) => [
-			filter( select( 'core/notices' ).getNotices(), {
-				id: UnpublishedChangesNotice.ID,
-			} ).length > 0,
-			select( 'core/block-editor' ).getBlocks(),
-			select( STORE_NAME ).isEditorSaving(),
-			select( STORE_NAME ).isEditorContentSaved(),
-		]
-	);
+	const [
+		canRestoreDraft,
+		editorContent,
+		isSaving,
+		isSaved,
+		projectTitle,
+	] = useSelect( ( select ) => [
+		filter( select( 'core/notices' ).getNotices(), {
+			id: UnpublishedChangesNotice.ID,
+		} ).length > 0,
+		select( 'core/block-editor' ).getBlocks(),
+		select( STORE_NAME ).isEditorSaving(),
+		select( STORE_NAME ).isEditorContentSaved(),
+		select( STORE_NAME ).getEditorTitle(),
+	] );
 
 	const saveProject = () =>
-		saveEditorContent( project.id, parse( serialize( editorContent ) ) );
+		saveEditorContent( project.id, parse( serialize( editorContent ) ), {
```

Comment:
This actually already was a drawback of us not having a reference of the to-be-saved changes - we have to extract blocks manually anytime we want to call `saveEditorContent`.  
Exactly the problem that having the timer in the state layer would solve.

Reply:
Oh, so this is like "pre-existing condition" sort of speak.